### PR TITLE
refactor(monorepo): move superset-ui to superset(stage 2)

### PR DESCRIPTION
**Pull request by zhaoyongjie**
### SUMMARY

This PR implement [SIP-58](https://github.com/apache/superset/issues/13013), move superset-ui to this repo. NOTICE: Please merge [relocation-PR](https://github.com/apache/superset/pull/17445) before merging this PR. 
1. Move `packages`, `plugins`, `scripts` directories into `superset_frontend`
2. All remaining files are saved in temporary_superset_ui
3. add `packages/*` and `plugins/*` to lerna packages
4. refactor superset-generator unit test
5. add `babel-extract.sh`. `packages` and `plugins` is appended in `babel.cfg`

image

-------------------------------------
To ensure that every PR passes the CI, the monorepo-related PR is split into multiple stages. 
1. https://github.com/apache/superset/pull/17427
2. https://github.com/apache/superset/pull/17445 
3. https://github.com/apache/superset/pull/17437 (this PR)

### CODE REVIEW

File: superset-frontend/jest.config.js [hunk 1]
```diff
@@ -16,33 +16,64 @@
  * specific language governing permissions and limitations
  * under the License.
  */
+
 module.exports = {
-  testRegex: '(\\/spec|\\/src)\\/.*(_spec|\\.test)\\.(j|t)sx?$',
+  testRegex: '\\/(spec|src|plugins|packages)\\/.*(_spec|\\.test)\\.[jt]sx?$',
+  testPathIgnorePatterns: [
+    'packages\\/generator-superset\\/.*',
```
Comment (villebro) [inline superset-frontend/jest.config.js:23]: should we add a TODO here to figure out why this was crashing tests?

**Review by villebro — COMMENTED** [review]
A few comments. In general I find this works very well, amazing work! 🎉  There will inevitably be some todos that will require follow-up PRs, but I think this is already a big improvement over the current development workflow, so feeling inclined to approve this to unblock plugin development.

File: superset/translations/babel.cfg [hunk 2]
```diff
@@ -14,12 +14,18 @@
...
-[javascript: superset-frontend/src/**.js]
-[javascript: superset-frontend/src/**.jsx]
-[javascript: superset-frontend/src/**.ts]
-[javascript: superset-frontend/src/**.tsx]
+[javascript: superset-frontend/src/**.js?]
+[javascript: superset-frontend/src/**.ts?]
+[javascript: superset-frontend/packages/**.js?]
+[javascript: superset-frontend/packages/**.ts?]
+[javascript: superset-frontend/plugins/**.js?]
+[javascript: superset-frontend/plugins/**.ts?]
```
[the remark this answers is not in the document]
Comment (zhaoyongjie) [reply superset/translations/babel.cfg]: Nice catch!

File: .codecov.yml [hunk 3]
```diff
@@ -14,6 +15,17 @@ coverage:
         # project coverage decrease:
         target: auto
         threshold: 0%
+      core-packages-ts:
+        target: 95%
```
[the remark this answers is not in the document]
Comment (zhaoyongjie) [reply .codecov.yml:19]: Thanks for the reminder.

File: superset-frontend/jest.config.js (hunk 1 above)
Comment (zhaoyongjie) [reply superset-frontend/jest.config.js:23]: I added a step in the [workflow](https://github.com/apache/superset/pull/17552/commits/fcbcca61b6662f9e63161e1184371392c489ef38) of frontend for this unit test.

**Review by geido — APPROVED** [review]
This is a giant leap. Thanks for all the hard work!

**Comment by stephenLYZ** [comment]
Tested in local. LGTM. Thanks for the huge improvement!

**Review by ktmud — APPROVED** [review]
This is really cool! Thanks for the hard work. I'm wondering if we can split CI for plugins into separate workflows just to make it easier to debug.

**Review by kgabryje — APPROVED** [review]
Tested manually, works great. Amazing work!

**Review by villebro — APPROVED** [review]
What an epic series of PRs! Amazing work @zhaoyongjie ! 🎉🎉🎉

**Pull request merged**